src/meta-schema/meta-schema.ts

```diff
@@ -667,6 +724,24 @@ export function getMetaSchema(project: Project): GraphQLSchema {
         );
     }
 
+    function getTypeLabelI18n(type: {}): Record<string, string | null> {
+        return mapValues(
+            model.i18n.getTypeI18n(type as Type),
+            (typeI18n) => typeI18n.label ?? null,
+        );
+    }
```

Comment:
You could move these methods into the actual `model/implementation` classes and make them getters, e.g. `get label(): Record<string, string>`, and annotate that with `@memorize()`. Then the lookups, which can happen quite frequently, wouldn't need to do all this map-wrangling. You also wouldn't need any custom resolvers at all.

While `null` values are technically allowed in `I18nString`, they are discarded in the scalar implementation, which requires cloning the value - so it would be better to omit these properties in the first place (and before the `@memorize()`).

Reply:
Ah yes, it is not really necessary to have functions here since graphql has default field-resolvers and `RootEntityType` etc. is already resolved to the concrete implementation 👍 
I put the logic in the respective base classes  (`Field`, `EnumValue` and `TypeBase`).